commit 248725a5ef70b0fb81d4b91b6b34188b9fa90013
Author: clayderhua <clayderhua@gmail.com>
Date:   Fri May 26 23:03:21 2023 +0800

    [ALL][NET] driver: Add the setting of lan LED for the RTL8211F

20260313: update for lf-6.18.2-1.0.0

Observation: adapted from https://github.com/ADVANTECH-Corp/linux-imx.git

Upstream-Status: Inappropriate [other]
Signed-off-by: F. Aragon <francisco.aragon@advantech.de>

diff --git a/drivers/net/phy/realtek/realtek_main.c b/drivers/net/phy/realtek/realtek_main.c
index c21844cbed03..270d012654ea 100644
--- a/drivers/net/phy/realtek/realtek_main.c
+++ b/drivers/net/phy/realtek/realtek_main.c
@@ -181,9 +181,28 @@ struct rtl821x_priv {
 	u16 iner;
 };
 
+static void rtl8211f_phy_fixup(struct phy_device *dev)
+{
+	struct device_node *node = dev->mdio.dev.of_node;
+
+	msleep(200);
+	phy_write(dev, 0x1f, 0x0d04);
+	/*PHY LED OK*/
+	phy_write(dev, 0x10, 0xa050);
+	phy_write(dev, 0x11, 0x0000);
+	phy_write(dev, 0x1f, 0x0000);
+}
+
+
 static int rtl821x_read_page(struct phy_device *phydev)
 {
-	return __phy_read(phydev, RTL821x_PAGE_SELECT);
+	int ret = __phy_read(phydev, RTL821x_PAGE_SELECT);
+
+	if (phydev && (0x001cc916 == phydev->phy_id)) {
+		rtl8211f_phy_fixup(phydev);
+	}
+
+	return ret;
 }
 
 static int rtl821x_write_page(struct phy_device *phydev, int page)
@@ -657,6 +676,10 @@ static int rtl8211f_config_init(struct phy_device *phydev)
 		return ret;
 	}
 
+	if (phydev && (0x001cc916 == phydev->phy_id)) {
+		rtl8211f_phy_fixup(phydev);
+	}
+
 	ret = rtl8211f_config_rgmii_delay(phydev);
 	if (ret)
 		return ret;
